pureMaterials/createPurematlib.py

```diff
@@ -6,73 +6,158 @@
 # -can be used for mixing homogenized regions
 # -generally impurities at <~1e-3 wt. percent (10 wppm) are removed from materials
 #   (except SS-316 steels may contain boron impurity)
-# 
+#
 #
 import os
+from typing import Dict, Union, Optional
 from pyne import material
 from pyne.material import Material, MultiMaterial
 from pyne.material_library import MaterialLibrary
+from pyne import nucname
 #
 #
 
-def make_mat(nucvec, density, citation, molecular_mass = None):
-    mat = Material(nucvec, density = density, metadata = {'citation' : citation})
+
+def enrich(
+    nucvec: Dict[int, float], old_key: int, new_values: Dict[int, float]
+) -> Dict[int, float]:
+    """Update nuclear vector by replacing an old key with new key-value pairs."""
+
+    if old_key in nucvec:
+        fract = nucvec.pop(old_key)
+        for key, value in new_values.items():
+            nucvec[key] = value * fract
+    else:
+        found_key = None
+        for key in list(nucvec.keys()):
+            if nucname.id(old_key) == nucname.id(key):
+                found_key = key
+                break
+
+        if found_key is not None:
+            fract = nucvec.pop(found_key)
+            for new_key, value in new_values.items():
+                nucvec[new_key] = value * fract  # check if percent works also
+
+    return nucvec
+
+
+def make_mat(
+    nucvec: Dict[int, float],
+    density: float,
+    citation: str,
+    molecular_mass: Optional[float] = None,
+    mass_enrichment: Optional[Dict[int, Dict[int, float]]] = None,
+) -> Material:
+    """Create a Material object from nuclear vector data."""
+    if mass_enrichment is not None:
+        for isotope, enrichment_vector in mass_enrichment.items():
+            enriched_mat = Material(
+                {key: value for key, value in enrichment_vector.items()}
+            )
+            nucvec = enrich(nucvec, isotope, enriched_mat.comp)
```

Comment:
I think your data structure keys on elements, not isotopes, right?
```suggestion
        for element, enrichment_vector in mass_enrichment.items():
            enriched_mat = Material(
                {key: value for key, value in enrichment_vector.items()}
            )
            nucvec = enrich(nucvec, element, enriched_mat.comp)
```